### DISCUSSION TITLE
Unexpected Latency Issue

### DISCUSSION THREAD

**Discussion by Mukhammadali** [supabase/supabase#16397] [Questions]
Hi, I chose Central Europe Frankfurt for the region on Supabase Cloud (free tier), and querying 20 simple rows (Countries table from sample SQL snippets) is taking almost 1 sec from Uzbekistan (Central Asia). I self-hosted the Supabase in the same region in Frankfurt on Vultr service and it is significantly faster for the same query with an average speed of 250-300ms. 

I debugged and noticed that the response was coming from Cloudflare while hitting the rest API and the Cloudflare server location was US, so I switched my VPN on to New York, and then the latency dropped from 1 sec to around 400-500ms. I tested from different internet providers and different devices,  yet the same high latency issue.  Is there any way to solve this from the Supabase side? Or is it a Cloudflare issue?

**Comment by Mukhammadali** [comment]
@kiwicopple

**Comment by prathap-puthran** [comment]
exactly same thing noticed.
